kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -458,6 +458,143 @@ function getReplicaCount {
 
 }
 
+#
+# Function to generate the domain restartVersion by incrementing the
+# existing value. If the restartVersion doesn't exist or the restartVersion
+# value is non-numeric, then return '1'.
+# $1 - Domain resource in json format
+# $2 - Return value containing the restart version.
+#
+function generateDomainRestartVersion {
+  local domainJson=$1
+  local __result=$2
+  local __restartVersion=""
+
+  eval $__result=""
+  __restartVersion=$(echo ${domainJson} | jq -cr .spec.restartVersion)
+  if ! [[ "$__restartVersion" =~ ^[0-9]+$ ]] ; then
+   __restartVersion=0
+  fi
+  __restartVersion=$((__restartVersion+1))
+  eval $__result=${__restartVersion}
+}
+
+#
+# Function to generate the domain introspectVersion by incrementing the
+# existing value. If the introspectVersion doesn't exist or the introspectVersion
+# value is non-numeric, then return '1'.
+# $1 - Domain resource in json format
+# $2 - Return value containing the introspect version.
+#
+function generateDomainIntrospectVersion {
+  local domainJson=$1
+  local __result=$2
+  local __introspectVersion=""
+
+  eval $__result=""
+  __introspectVersion=$(echo ${domainJson} | jq -cr .spec.introspectVersion)
+  if ! [[ "$__introspectVersion" =~ ^[0-9]+$ ]] ; then
+   __introspectVersion=0
+  fi
+  __introspectVersion=$((__introspectVersion+1))
+  eval $__result=${__introspectVersion}
+}
+
+#
+# Function to generate the cluster restartVersion by incrementing the
+# existing value. If the restartVersion doesn't exist at the cluster level, 
+# then it increments the domain level restartVersion value. If the effective 
```

Comment:
It is not clear to me if the script actually increments the domain level restartVersion or just set the cluster restartVersion with the incremented domain level restartVersion.  I think it is latter, but not sure. If so, the wording needs to be clearer.

Reply:
It is probably better in this case not to take the incremented domain level restartVersion, instead just set the cluster level restartVersion to 1.

Reply:
I'm not sure if the cluster will roll if the domain level restartVersion is 1 and then the cluster level restartVersion is also updated to 1.